quantiles/test/quantiles_sketch_test.cpp

```diff
@@ -0,0 +1,899 @@
+    REQUIRE(sketch2.get_normalized_rank_error(true) == sketch1.get_normalized_rank_error(true));
+    REQUIRE(sketch2.get_quantile(0.5) == sketch1.get_quantile(0.5));
+    REQUIRE(sketch2.get_rank(std::to_string(0)) == sketch1.get_rank(std::to_string(0)));
+    REQUIRE(sketch2.get_rank(std::to_string(n)) == sketch1.get_rank(std::to_string(n)));
+  }
+
+  SECTION("sketch of strings, single item, bytes") {
+    quantiles_string_sketch sketch1(64, 0);
+    sketch1.update("a");
+    auto bytes = sketch1.serialize();
+    REQUIRE(bytes.size() == sketch1.get_serialized_size_bytes());
+    auto sketch2 = quantiles_string_sketch::deserialize(bytes.data(), bytes.size(), serde<std::string>(), 0);
+    REQUIRE(bytes.size() == sketch2.get_serialized_size_bytes());
+  }
+
+  SECTION("copy") {
+    quantiles_sketch<int> sketch1;
+    const int n(1000);
+    for (int i = 0; i < n; i++) sketch1.update(i);
+
+    // copy constructor
+    quantiles_sketch<int> sketch2(sketch1);
+    for (int i = 0; i < n; i++) {
+      REQUIRE(sketch2.get_rank(i) == sketch1.get_rank(i));
+    }
+
+    // copy assignment
+    quantiles_sketch<int> sketch3;
+    sketch3 = sketch1;
+    for (int i = 0; i < n; i++) {
+      REQUIRE(sketch3.get_rank(i) == sketch1.get_rank(i));
+    }
+  }
+
+  SECTION("move") {
+    quantiles_sketch<int> sketch1;
+    const int n(100);
+    for (int i = 0; i < n; i++) sketch1.update(i);
+
+    // move constructor
+    quantiles_sketch<int> sketch2(std::move(sketch1));
+    for (int i = 0; i < n; i++) {
+      REQUIRE(sketch2.get_rank(i) == (double) i / n);
+    }
+
+    // move assignment
+    quantiles_sketch<int> sketch3;
+    sketch3 = std::move(sketch2);
+    for (int i = 0; i < n; i++) {
+      REQUIRE(sketch3.get_rank(i) == (double) i / n);
+    }
+  }
+/*
+  SECTION("max serialized size arithmetic type") {
+    REQUIRE(quantiles_sketch<float>::get_max_serialized_size_bytes(128, 10) == 1968);
+    REQUIRE(quantiles_sketch<float>::get_max_serialized_size_bytes(128, 100) == 2316);
+    REQUIRE(quantiles_sketch<float>::get_max_serialized_size_bytes(128, 1000) == 2440);
+    REQUIRE(quantiles_sketch<float>::get_max_serialized_size_bytes(256, 1000000) == 2800);
+    REQUIRE(quantiles_sketch<float>::get_max_serialized_size_bytes(256, 1000000000) == 3160);
+  }
+
+  SECTION("max serialized size non-arithmetic type") {
+    REQUIRE(quantiles_sketch<std::string>::get_max_serialized_size_bytes(200, 10, 4) == 1968);
+    REQUIRE(quantiles_sketch<std::string>::get_max_serialized_size_bytes(200, 100, 4) == 2316);
+    REQUIRE(quantiles_sketch<std::string>::get_max_serialized_size_bytes(200, 1000, 4) == 2440);
+    REQUIRE(quantiles_sketch<std::string>::get_max_serialized_size_bytes(200, 1000000, 4) == 2800);
+    REQUIRE(quantiles_sketch<std::string>::get_max_serialized_size_bytes(200, 1000000000, 4) == 3160);
+  }
+*/
+
+  // cleanup
+  if (test_allocator_total_bytes != 0) {
+    REQUIRE(test_allocator_total_bytes == 0);
+  }
+}
+
```

Comment:
I don't see any tests in the inclusive mode

Reply:
only get_rank() actually uses the value of inclusive.  only added direct tests of that, but feel free to indicate that more tests are needed

Reply:
Oops, also quantile calculator.. need to add that still.

Reply:
added

Reply:
get_PMF and get_CDF also can be inclusive or not (however they are implemented in terms of get_rank)